# SYNTAX TEST "Packages/Diff/Diff.sublime-syntax"

--- Path to Original File
#^^^^^^^^^^^^^^^^^^^^^^^^ meta.header.from-file.diff
# <-                      punctuation.definition.from-file.diff
#^^                       punctuation.definition.from-file.diff

+++ Path to Modified File
#^^^^^^^^^^^^^^^^^^^^^^^^ meta.header.to-file.diff
# <-                      punctuation.definition.to-file.diff
#^^                       punctuation.definition.to-file.diff

28a211
#^^^^^ meta.range.normal.diff

@@ -2,8 +2,11 @@
#^^^^^^^^^^^^^^^ meta.range.unified.diff
# <-             punctuation.definition.range.diff
#^               punctuation.definition.range.diff
#  ^^^^^^^^^^    meta.toc-list.line-number.diff
#             ^^ punctuation.definition.range.diff

--- Range ----
#^^^^^^^^^^^^^ meta.range.context.diff
# <-           punctuation.definition.range.diff
#^^            punctuation.definition.range.diff
#         ^^^^ punctuation.definition.range.diff

*** Range ****
#^^^^^^^^^^^^^ meta.range.context.diff
# <-           punctuation.definition.range.diff
#^^            punctuation.definition.range.diff
#         ^^^^ punctuation.definition.range.diff

***************
#^^^^^^^^^^^^^^ meta.separator.diff punctuation.definition.separator.diff

****************
#^^^^^^^^^^^^^^^ -meta.separator.diff punctuation.definition.separator.diff

===================================================================
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.separator.diff punctuation.definition.separator.diff

====================================================================
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ -meta.separator.diff punctuation.definition.separator.diff

---
#^^ meta.separator.diff
#^^ punctuation.definition.separator.diff

----
#^^^ -meta.separator.diff
#^^^ -punctuation.definition.separator.diff

Plain Text
#^^^^^^^^^ source.diff

+ Addition
# <-       punctuation.definition.inserted.diff
# ^^^^^^^^ markup.inserted.diff

> Addition
# <-       punctuation.definition.inserted.diff
# ^^^^^^^^ markup.inserted.diff

- Deletion
# <-       punctuation.definition.deleted.diff
# ^^^^^^^^ markup.deleted.diff

< Deletion
# <-       punctuation.definition.deleted.diff
# ^^^^^^^^ markup.deleted.diff

! Modified
# <-       punctuation.definition.changed.diff
# ^^^^^^^^ markup.changed.diff

Index: value
#^^^^^^^^^^^ meta.index.diff
#    ^       punctuation.separator.key-value.diff
#      ^^^^^ meta.toc-list.file-name.diff
